source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -38,34 +40,58 @@ void UdpStatsdSink::WriterImpl::write(const std::string& message) {
   Network::Utility::writeToSocket(*io_handle_, &slice, 1, nullptr, *parent_.server_address_);
 }
 
+void UdpStatsdSink::WriterImpl::writeBuffer(Buffer::Instance* data) {
+  Network::Utility::writeToSocket(*io_handle_, *data, nullptr, *parent_.server_address_);
+}
+
 UdpStatsdSink::UdpStatsdSink(ThreadLocal::SlotAllocator& tls,
                              Network::Address::InstanceConstSharedPtr address, const bool use_tag,
-                             const std::string& prefix)
+                             const std::string& prefix, uint64_t buffer_size)
     : tls_(tls.allocateSlot()), server_address_(std::move(address)), use_tag_(use_tag),
-      prefix_(prefix.empty() ? Statsd::getDefaultPrefix() : prefix) {
+      prefix_(prefix.empty() ? Statsd::getDefaultPrefix() : prefix), buffer_size_(buffer_size) {
   tls_->set([this](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
     return std::make_shared<WriterImpl>(*this);
   });
 }
 
 void UdpStatsdSink::flush(Stats::MetricSnapshot& snapshot) {
-  Writer& writer = tls_->getTyped<Writer>();
+  auto buffer = Buffer::OwnedImpl();
   for (const auto& counter : snapshot.counters()) {
     if (counter.counter_.get().used()) {
-      writer.write(absl::StrCat(prefix_, ".", getName(counter.counter_.get()), ":", counter.delta_,
-                                "|c", buildTagStr(counter.counter_.get().tags())));
+      auto counterStr =
```

Comment:
`std::string counter_str`

Please read https://google.github.io/styleguide/cppguide.html#Type_deduction for details as to why we don't use `auto` here.

And see https://github.com/envoyproxy/envoy/blob/master/STYLE.md for variable-name casing syntax.

Reply:
Fixed! And thank you for the links! Regarding the case styles, I'm not seeing anything specifically for variable names.

Reply:
it is a bit confusing because Envoy follows the Google C++ style guide except for exceptions (sic) :)

https://google.github.io/styleguide/cppguide.html#Variable_Names is the source of truth though for local variables in a function.